codegenerator/cli/templates/static/codegen/src/bindings/HyperSyncClient.res

```diff
@@ -10,180 +10,342 @@ type cfg = {
+     *  pagination.
+     */
+    @as("toBlock")
+    toBlockExclusive?: int,
+    /**
+     * List of log selections, these have an or relationship between them, so the query will return logs
+     * that match any of these selections.
+     */
+    logs?: array<logSelection>,
+    /**
+     * List of transaction selections, the query will return transactions that match any of these selections and
+     *  it will return transactions that are related to the returned logs.
+     */
+    transactions?: array<transactionSelection>,
+    /**
+     * List of trace selections, the query will return traces that match any of these selections and
+     *  it will re turn traces that are related to the returned logs.
+     */
+    traces?: array<traceSelection>,
+    /** List of block selections, the query will return blocks that match any of these selections */
+    blocks?: array<blockSelection>,
+    /**
+     * Field selection. The user can select which fields they are interested in, requesting less fields will improve
+     *  query execution time and reduce the payload size so the user should always use a minimal number of fields.
+     */
     fieldSelection: fieldSelection,
+    /**
+     * Maximum number of blocks that should be returned, the server might return more blocks than this number but
+     *  it won't overshoot by too much.
+     */
+    maxNumBlocks?: int,
+    /**
+     * Maximum number of transactions that should be returned, the server might return more transactions than this number but
+     *  it won't overshoot by too much.
+     */
+    maxNumTransactions?: int,
+    /**
+     * Maximum number of logs that should be returned, the server might return more logs than this number but
+     *  it won't overshoot by too much.
+     */
     maxNumLogs?: int,
+    /**
+     * Maximum number of traces that should be returned, the server might return more traces than this number but
+     *  it won't overshoot by too much.
+     */
+    maxNumTraces?: int,
+    /**
+     * Selects join mode for the query,
+     * Default: join in this order logs -> transactions -> traces -> blocks
+     * JoinAll: join everything to everything. For example if logSelection matches log0, we get the
+     * associated transaction of log0 and then we get associated logs of that transaction as well. Applites similarly
+     * to blocks, traces.
+     * JoinNothing: join nothing.
+     */
+    joinMode?: joinMode,
   }
 }
 
 module ResponseTypes = {
-  //Note all fields marked as "nullable" are not explicitly null since
-  //the are option fields and nulls will be deserialized to option when
-  //in an optional field with spice
-  type blockData = {
+  type withdrawal = {
+    index?: string,
+    validatorIndex?: string,
+    address?: Address.t,
+    amount?: string,
+  }
+
+  type block = {
     number?: int,
     hash?: string,
     parentHash?: string,
-    nonce?: int, //nullable
+    nonce?: bigint,
```

Comment:
Maybe keep the `nullable` comment? And if it's returned as `null` won't it break things?